From a96257af783f1d2c35a957466856e62ebf82bcad Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 3 May 2011 17:58:28 +0200
Subject: [PATCH] vconsole: use open_terminal() instead of open()

---
 src/vconsole-setup.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/vconsole-setup.c b/src/vconsole-setup.c
index 1be260b..68ebac9 100644
--- a/src/vconsole-setup.c
+++ b/src/vconsole-setup.c
@@ -176,7 +176,7 @@ int main(int argc, char **argv) {
         else
                 vc = "/dev/tty0";
 
-        if ((fd = open(vc, O_RDWR|O_CLOEXEC)) < 0) {
+        if ((fd = open_terminal(vc, O_RDWR|O_CLOEXEC)) < 0) {
                 log_error("Failed to open %s: %m", vc);
                 goto finish;
         }
-- 
1.7.5.2

