From 3fdbc8205885f117b7dea289b44217310663e731 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 21 Aug 2014 16:10:37 +0200
Subject: [PATCH] CODING_STYLE: document that we don't break lines at 80ch

---
 CODING_STYLE | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CODING_STYLE b/CODING_STYLE
index ca3b5183f9..a3fc26c1e1 100644
--- a/CODING_STYLE
+++ b/CODING_STYLE
@@ -1,6 +1,9 @@
-
 - 8ch indent, no tabs
 
+- Don't break code lines too eagerly. We do *not* force line breaks at
+  80ch, all of today's screens should be much larger than that. But
+  then again, don't overdo it, ~140ch should be enough really.
+
 - Variables and functions *must* be static, unless they have a
   prototype, and are supposed to be exported.
 
