From 64a94f1e45890eaeb5880dbd2044265e17023067 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Sun, 15 Sep 2013 07:29:25 +0200
Subject: [PATCH] README: add SCSI BSG option

---
 README | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/README b/README
index a16f342..b39cd37 100644
--- a/README
+++ b/README
@@ -64,6 +64,10 @@ REQUIREMENTS:
         Mount and bind mount handling might require it:
           CONFIG_FHANDLE
 
+        Support for some SCSI devices serial number retrieval, to
+        create additional symlinks in /dev/disk/ and /dev/tape:
+          CONFIG_BLK_DEV_BSG
+
         Optional but strongly recommended:
           CONFIG_IPV6
           CONFIG_AUTOFS4_FS
