From 55cdcbacf70f05a40a155af24f6d2da6b478cba6 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 21 Aug 2014 17:20:00 +0200
Subject: [PATCH] update TODO

---
 TODO | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/TODO b/TODO
index cbd8384b4a..868518ab75 100644
--- a/TODO
+++ b/TODO
@@ -38,9 +38,6 @@ Features:
   for "systemctl suspend" to finish to know when the suspending is
   complete.
 
-* sd_notify("SHUTDOWN=1") to fix a dbus activation race.
-  http://lists.freedesktop.org/archives/systemd-devel/2014-July/020983.html
-
 * merge ~/.local/share and ~/.local/lib into one similar /usr/lib and /usr/share....
 
 * make "systemctl suspend" block until we are back from suspend
@@ -603,9 +600,7 @@ Features:
 
 * make sure systemd-ask-password-wall does not shutdown systemd-ask-password-console too early
 
-* support sd_notify() style notification when reload begins (RELOADING=1), reload is finished (READY=1), and add ReloadSignal= then to use in combination
-
-* support sd_notify() style notification when shutting down, to make auto-exit bus services work (STOPPING=1)
+* add ReloadSignal= for configuring a reload signal to use
 
 * verify that the AF_UNIX sockets of a service in the fs still exist
   when we start a service in order to avoid confusion when a user
