From 337ce7442a0602116c6253ebf202bd34f675f627 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 21 Aug 2014 19:12:43 +0200
Subject: [PATCH] update TODO

---
 TODO | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/TODO b/TODO
index 16b61d045c..3073f3a5a2 100644
--- a/TODO
+++ b/TODO
@@ -521,10 +521,6 @@ Features:
 
 * properly handle loop back mounts via fstab, especially regards to fsck/passno
 
-* allow services with no ExecStart= but with an ExecStop=
-
-* dracut-shutdown needs to be ordered before unmounting /boot
-
 * initialize the hostname from the fs label of /, if /etc/hostname does not exist?
 
 * rename "userspace" to "core-os"
