From d74f9e8e8a3dcddb043ef193e4bb14f58efa095f Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 22 Aug 2014 18:10:22 +0200
Subject: [PATCH] update TODO

---
 TODO | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/TODO b/TODO
index 3073f3a5a2..0fcd3a0b8e 100644
--- a/TODO
+++ b/TODO
@@ -24,6 +24,9 @@ External:
 
 Features:
 
+* apply start timeout during the "initializing" manager state only,
+  instead of both "initializing" and "starting".
+
 * journald: make use of uid-range.h to managed uid ranges to split
   journals in.
 
@@ -138,8 +141,6 @@ Features:
 * For timer units: add some mechanisms so that timer units that trigger immediately on boot do not have the services
   they run added to the initial transaction and thus confuse Type=idle.
 
-* Add timeout to early-boot, and shut down the system if it is hit. Solves the laptop-in-bag problem and is useful for embedded cases
-
 * Run most system services with cgroupfs read-only and procfs with a more secure mode (doesn't work, since the hidepid= option is per-pid-namespace, not per-mount)
 
 * sd-event: generate a failure of a default event loop is executed out-of-thread
