From 42aeb14a4a0fa7d43da96a8ed0fb0e180a2dd5c8 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 26 Aug 2014 03:59:05 +0200
Subject: [PATCH] update TODO

---
 TODO | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/TODO b/TODO
index 0fcd3a0b8e..471d3b29bc 100644
--- a/TODO
+++ b/TODO
@@ -24,8 +24,15 @@ External:
 
 Features:
 
+* remove multi-seat-x now
+
+* refcounting in sd-resolve is borked
+
+* exponential backoff in timesyncd and resolved when we cannot reach a server
+
 * apply start timeout during the "initializing" manager state only,
-  instead of both "initializing" and "starting".
+  instead of both "initializing" and "starting". maybe rename the
+  timeout to "initialization-timeout" then or so?
 
 * journald: make use of uid-range.h to managed uid ranges to split
   journals in.
