From d0bd01f51f5b7575d867208f27dbe4dc9299da7f Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Wed, 4 Apr 2012 14:08:37 +0200
Subject: [PATCH] build-sys: add a few missing headers (cherry picked from
 commit 8d62c69d9e4a3a0013310735ee2c1e746abea29a)

---
 Makefile.am |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 710b58c..2079298 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -585,13 +585,17 @@ libsystemd_core_la_SOURCES = \
 	src/install.c \
 	src/install.h \
 	src/specifier.c \
+	src/specifier.h \
 	src/namespace.c \
+	src/namespace.h \
 	src/unit-name.c \
+	src/unit-name.h \
 	src/tcpwrap.c \
-	src/sd-daemon.c \
+	src/tcpwrap.h \
 	src/cgroup-attr.c \
-	src/sd-id128.c \
 	src/cgroup-attr.h \
+	src/sd-daemon.c \
+	src/sd-id128.c \
 	src/macro.h \
 	src/def.h \
 	src/ioprio.h \
