From daa05349dfefb12638c96e034c11be613bdc39b7 Mon Sep 17 00:00:00 2001
From: Ansgar Burchardt <ansgar@43-1.org>
Date: Tue, 26 Aug 2014 00:19:54 +0200
Subject: [PATCH] NEWS: Fix typos.

---
 NEWS | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/NEWS b/NEWS
index 7dad765a33..2fca2cdc93 100644
--- a/NEWS
+++ b/NEWS
@@ -42,8 +42,8 @@ CHANGES WITH 216:
 
         * systemd-resolved now includes a caching DNS stub resolver
           and a complete LLMNR name resolution implementation. A new
-          NSS module "nss-resolve" has been added which make be used
-          of glibc's own "nss-dns" to resolve hostnames via
+          NSS module "nss-resolve" has been added which can be used
+          instead of glibc's own "nss-dns" to resolve hostnames via
           systemd-resolved. Hostnames, addresses and arbitrary RRs may
           be resolved via systemd-resolved D-Bus APIs. In contrast to
           the glibc internal resolver systemd-resolved is aware of
@@ -606,7 +606,7 @@ CHANGES WITH 214:
 
         * Access modes specified in tmpfiles snippets may now be
           prefixed with "~", which indicates that they shall be masked
-          by whether the existing file or directly is currently
+          by whether the existing file or directory is currently
           writable, readable or executable at all. Also, if specified,
           the sgid/suid/sticky bits will be masked for all
           non-directories.
