From 00b333bb1039809a42ba4c7f25ba85f68766477d Mon Sep 17 00:00:00 2001
From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Wed, 27 Aug 2014 23:23:10 +0200
Subject: [PATCH] terminal: remove unused variable

---
 src/libsystemd-terminal/idev-keyboard.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/libsystemd-terminal/idev-keyboard.c b/src/libsystemd-terminal/idev-keyboard.c
index 647aade932..7ab4db2cf7 100644
--- a/src/libsystemd-terminal/idev-keyboard.c
+++ b/src/libsystemd-terminal/idev-keyboard.c
@@ -312,7 +312,6 @@ static int kbdctx_locale_props_changed_fn(sd_bus *bus,
                                           sd_bus_message *signal,
                                           void *userdata,
                                           sd_bus_error *ret_err) {
-        _cleanup_bus_message_unref_ sd_bus_message *m = NULL;
         kbdctx *kc = userdata;
         int r;
 
