From 4df5c00b6e5de8733f3bb33ee7980fad1a498789 Mon Sep 17 00:00:00 2001
From: Umut Tezduyar Lindskog <umut.tezduyar@axis.com>
Date: Tue, 2 Sep 2014 12:31:49 +0200
Subject: [PATCH] build-sys: configure option to disable hibernation

---
 Makefile.am  | 52 ++++++++++++++++++++++++++++++++--------------------
 configure.ac |  6 ++++++
 2 files changed, 38 insertions(+), 20 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 1991fd0e3b..58e5ce6c54 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -378,15 +378,13 @@ rootlibexec_PROGRAMS = \
 	systemd-sleep \
 	systemd-bus-proxyd \
 	systemd-socket-proxyd \
-	systemd-update-done \
-	systemd-hibernate-resume
+	systemd-update-done
 
 systemgenerator_PROGRAMS = \
 	systemd-getty-generator \
 	systemd-fstab-generator \
 	systemd-system-update-generator \
-	systemd-debug-generator \
-	systemd-hibernate-resume-generator
+	systemd-debug-generator
 
 dist_bashcompletion_DATA = \
 	shell-completion/bash/busctl \
@@ -453,7 +451,6 @@ dist_systemunit_DATA = \
 	units/network-online.target \
 	units/nss-lookup.target \
 	units/nss-user-lookup.target \
-	units/hibernate.target \
 	units/hybrid-sleep.target \
 	units/poweroff.target \
 	units/reboot.target \
@@ -511,7 +508,6 @@ nodist_systemunit_DATA = \
 	units/emergency.service \
 	units/rescue.service \
 	units/user@.service \
-	units/systemd-hibernate.service \
 	units/systemd-hybrid-sleep.service \
 	units/systemd-suspend.service \
 	units/systemd-halt.service \
@@ -530,8 +526,7 @@ nodist_systemunit_DATA = \
 	units/initrd-udevadm-cleanup-db.service \
 	units/initrd-switch-root.service \
 	units/systemd-nspawn@.service \
-	units/systemd-update-done.service \
-	units/systemd-hibernate-resume@.service
+	units/systemd-update-done.service
 
 dist_userunit_DATA = \
 	units/user/basic.target \
@@ -569,7 +564,6 @@ EXTRA_DIST += \
 	units/systemd-fsck-root.service.in \
 	units/user@.service.in \
 	units/debug-shell.service.in \
-	units/systemd-hibernate.service.in \
 	units/systemd-hybrid-sleep.service.in \
 	units/systemd-suspend.service.in \
 	units/quotaon.service.in \
@@ -578,8 +572,7 @@ EXTRA_DIST += \
 	units/initrd-udevadm-cleanup-db.service.in \
 	units/initrd-switch-root.service.in \
 	units/systemd-nspawn@.service.in \
-	units/systemd-update-done.service.in \
-	units/systemd-hibernate-resume@.service.in
+	units/systemd-update-done.service.in
 
 CLEANFILES += \
 	units/console-shell.service.m4 \
@@ -2112,14 +2105,6 @@ systemd_delta_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
-systemd_hibernate_resume_SOURCES = \
-	src/hibernate-resume/hibernate-resume.c
-
-systemd_hibernate_resume_LDADD = \
-	libsystemd-internal.la \
-	libsystemd-shared.la
-
-# ------------------------------------------------------------------------------
 systemd_getty_generator_SOURCES = \
 	src/getty-generator/getty-generator.c
 
@@ -2153,6 +2138,20 @@ systemd_system_update_generator_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
+if ENABLE_HIBERNATE
+systemgenerator_PROGRAMS += \
+        systemd-hibernate-resume-generator
+
+rootlibexec_PROGRAMS += \
+        systemd-hibernate-resume
+
+systemd_hibernate_resume_SOURCES = \
+        src/hibernate-resume/hibernate-resume.c
+
+systemd_hibernate_resume_LDADD = \
+        libsystemd-internal.la \
+        libsystemd-shared.la
+
 systemd_hibernate_resume_generator_SOURCES = \
 	src/resume-generator/resume-generator.c
 
@@ -2160,8 +2159,21 @@ systemd_hibernate_resume_generator_LDADD = \
 	libsystemd-label.la \
 	libsystemd-shared.la
 
-if ENABLE_EFI
+EXTRA_DIST += \
+        units/systemd-hibernate.service.in \
+        units/systemd-hibernate-resume@.service.in
+
+dist_systemunit_DATA += \
+        units/hibernate.target
+
+nodist_systemunit_DATA += \
+        units/systemd-hibernate.service \
+        units/systemd-hibernate-resume@.service
+
+endif
+
 # ------------------------------------------------------------------------------
+if ENABLE_EFI
 systemgenerator_PROGRAMS +=  \
 	systemd-efi-boot-generator
 
diff --git a/configure.ac b/configure.ac
index 543828c405..99c01d2487 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1168,6 +1168,12 @@ AS_IF([test "x$enable_manpages" != xno], [have_manpages=yes])
 AM_CONDITIONAL(ENABLE_MANPAGES, [test "x$have_manpages" = "xyes"])
 
 # ------------------------------------------------------------------------------
+AC_ARG_ENABLE(hibernate,
+       [AC_HELP_STRING([--disable-hibernate], [disable hibernation support])],
+       enable_hibernate=$enableval, enable_hibernate=yes)
+AM_CONDITIONAL(ENABLE_HIBERNATE, [test x$enable_hibernate = xyes])
+
+# ------------------------------------------------------------------------------
 AC_ARG_ENABLE(ldconfig,
        [AC_HELP_STRING([--disable-ldconfig], [disable ldconfig])],
        enable_ldconfig=$enableval, enable_ldconfig=yes)
