From 8e8af4cfc7fa373504a22e58966909161acfb72f Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <systemd@esmil.dk>
Date: Thu, 18 Sep 2014 15:24:43 +0200
Subject: [PATCH] shared/sparse-endian.h: add missing byteswap.h include

---
 src/shared/sparse-endian.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/shared/sparse-endian.h b/src/shared/sparse-endian.h
index eb4dbf3615..c913fda8c5 100644
--- a/src/shared/sparse-endian.h
+++ b/src/shared/sparse-endian.h
@@ -21,6 +21,7 @@
 #ifndef SPARSE_ENDIAN_H
 #define SPARSE_ENDIAN_H
 
+#include <byteswap.h>
 #include <endian.h>
 #include <stdint.h>
 
