From 303174638af7a8fcf6211fb99b8d07c9d270567e Mon Sep 17 00:00:00 2001
From: Daniel Mack <daniel@zonque.org>
Date: Fri, 19 Sep 2014 17:50:41 +0200
Subject: [PATCH] bus-policy: story mandatory items in right list

---
 src/bus-proxyd/bus-policy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bus-proxyd/bus-policy.c b/src/bus-proxyd/bus-policy.c
index d2eace9405..8676d31f62 100644
--- a/src/bus-proxyd/bus-policy.c
+++ b/src/bus-proxyd/bus-policy.c
@@ -332,7 +332,7 @@ static int file_load(Policy *p, const char *path) {
                                 if (policy_category == POLICY_CATEGORY_DEFAULT)
                                         LIST_PREPEND(items, p->default_items, i);
                                 else if (policy_category == POLICY_CATEGORY_MANDATORY)
-                                        LIST_PREPEND(items, p->default_items, i);
+                                        LIST_PREPEND(items, p->mandatory_items, i);
                                 else if (policy_category == POLICY_CATEGORY_USER) {
                                         const char *u = policy_user;
 
