From c3502b59ec4e58a877003050e6c2fc668eee3129 Mon Sep 17 00:00:00 2001
From: Daniel Mack <daniel@zonque.org>
Date: Sat, 20 Sep 2014 16:34:31 +0200
Subject: [PATCH] bus_policy: set i->[ug]id_valid

---
 src/bus-proxyd/bus-policy.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/bus-proxyd/bus-policy.c b/src/bus-proxyd/bus-policy.c
index eed542d8f8..2c4708dd77 100644
--- a/src/bus-proxyd/bus-policy.c
+++ b/src/bus-proxyd/bus-policy.c
@@ -364,6 +364,7 @@ static int file_load(Policy *p, const char *path) {
 
                                                 first = hashmap_get(p->user_items, UINT32_TO_PTR(i->uid));
                                                 item_append(i, &first);
+                                                i->uid_valid = true;
 
                                                 r = hashmap_replace(p->user_items, UINT32_TO_PTR(i->uid), first);
                                                 if (r < 0) {
@@ -395,6 +396,7 @@ static int file_load(Policy *p, const char *path) {
 
                                                 first = hashmap_get(p->group_items, UINT32_TO_PTR(i->gid));
                                                 item_append(i, &first);
+                                                i->gid_valid = true;
 
                                                 r = hashmap_replace(p->group_items, UINT32_TO_PTR(i->gid), first);
                                                 if (r < 0) {
