From ed91202f1c237a41a3ee3754a4a1d37139d7f34f Mon Sep 17 00:00:00 2001
From: Daniel Mack <daniel@zonque.org>
Date: Sat, 20 Sep 2014 18:09:00 +0200
Subject: [PATCH] bus-policy: print numeric [gu]id in dump_items()

---
 src/bus-proxyd/bus-policy.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/bus-proxyd/bus-policy.c b/src/bus-proxyd/bus-policy.c
index 4bc575f3ab..e870fbc948 100644
--- a/src/bus-proxyd/bus-policy.c
+++ b/src/bus-proxyd/bus-policy.c
@@ -704,8 +704,8 @@ static void dump_items(PolicyItem *items, const char *prefix) {
 
                         user = uid_to_name(i->uid);
 
-                        printf("%sUser: %s\n",
-                               prefix, strna(user));
+                        printf("%sUser: %s (%d)\n",
+                               prefix, strna(user), i->uid);
                 }
 
                 if (i->gid_valid) {
@@ -713,8 +713,8 @@ static void dump_items(PolicyItem *items, const char *prefix) {
 
                         group = gid_to_name(i->gid);
 
-                        printf("%sGroup: %s\n",
-                               prefix, strna(group));
+                        printf("%sGroup: %s (%d)\n",
+                               prefix, strna(group), i->gid);
                 }
         }
 }
