From 10893a5cfa7d792ba171282c2ec46b85ed6aae0c Mon Sep 17 00:00:00 2001
From: Gustavo Sverzut Barbieri <gustavo.barbieri@intel.com>
Date: Thu, 25 Sep 2014 18:08:02 -0300
Subject: [PATCH] journal: build fix when LZ4 is enabled but XZ is not

---
 src/journal/journal-file.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/journal/journal-file.h b/src/journal/journal-file.h
index da2ef3b795..6b4bf0d5ae 100644
--- a/src/journal/journal-file.h
+++ b/src/journal/journal-file.h
@@ -78,7 +78,7 @@ typedef struct JournalFile {
 
         Hashmap *chain_cache;
 
-#ifdef HAVE_XZ
+#if defined(HAVE_XZ) || defined(HAVE_LZ4)
         void *compress_buffer;
         size_t compress_buffer_size;
 #endif
