From ef99aec4d25087dec995b3f00b6957dcee6b13e9 Mon Sep 17 00:00:00 2001
From: Gustavo Sverzut Barbieri <gustavo.barbieri@intel.com>
Date: Thu, 25 Sep 2014 18:13:19 -0300
Subject: [PATCH] only build and install systemd-bus-proxyd if --enable-kdbus

---
 Makefile.am | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 53128518b4..065d688ebb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -387,7 +387,6 @@ rootlibexec_PROGRAMS = \
 	systemd-ac-power \
 	systemd-sysctl \
 	systemd-sleep \
-	systemd-bus-proxyd \
 	systemd-socket-proxyd \
 	systemd-update-done
 
@@ -2428,6 +2427,10 @@ systemd_run_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
+if ENABLE_KDBUS
+rootlibexec_PROGRAMS += \
+	systemd-bus-proxyd
+
 systemd_bus_proxyd_SOURCES = \
 	src/bus-proxyd/bus-proxyd.c \
 	src/bus-proxyd/bus-policy.c \
@@ -2449,7 +2452,6 @@ bus-proxyd-uninstall-hook:
 INSTALL_EXEC_HOOKS += bus-proxyd-install-hook
 UNINSTALL_EXEC_HOOKS += bus-proxyd-uninstall-hook
 
-if ENABLE_KDBUS
 nodist_systemunit_DATA += \
 	units/systemd-bus-proxyd@.service
 
