From 2f88c8583aea6626c5f0a84cfc6de4b0a17f2d1c Mon Sep 17 00:00:00 2001
From: Gustavo Sverzut Barbieri <gustavo.barbieri@intel.com>
Date: Thu, 25 Sep 2014 18:12:03 -0300
Subject: [PATCH] do not install factory/etc/pam.d if --disable-pam

---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index bf860cdc00..1bdc91c239 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2048,9 +2048,11 @@ endif
 dist_factory_etc_DATA = \
 	factory/etc/nsswitch.conf
 
+if HAVE_PAM
 dist_factory_pam_DATA = \
 	factory/etc/pam.d/system-auth \
 	factory/etc/pam.d/other
+endif
 
 # ------------------------------------------------------------------------------
 if ENABLE_FIRSTBOOT
