From 56a7dd42d3a680691f417bd8a4c46ea1b6283ff5 Mon Sep 17 00:00:00 2001
From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Wed, 1 Oct 2014 23:22:02 +0200
Subject: [PATCH] gitignore: add test-set

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index b78a4cb4e1..cb1af8de5b 100644
--- a/.gitignore
+++ b/.gitignore
@@ -220,6 +220,7 @@
 /test-rtnl
 /test-rtnl-manual
 /test-sched-prio
+/test-set
 /test-sleep
 /test-socket-util
 /test-ssd
