From c7eff5ec06c3025d1ef14f955c653259ae7c615b Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 2 Oct 2014 20:36:23 +0200
Subject: [PATCH] update TODO

---
 TODO | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/TODO b/TODO
index 9ac6fac8cf..526926f862 100644
--- a/TODO
+++ b/TODO
@@ -26,6 +26,8 @@ External:
 
 Features:
 
+* busctl: add a pcap writer, using LINKTYPE_DBUS/231
+
 * man: maybe use the word "inspect" rather than "introspect"?
 
 * introduce machines.target to order after all nspawn instances
