From f45d32872cb65530b6c7b4818c40e917b44b2633 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 3 Oct 2014 22:08:23 -0400
Subject: [PATCH] build-sys: fix make distcheck

---
 Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 60011b7d98..cca949feff 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -712,7 +712,8 @@ man/systemd.directives.xml: $(top_srcdir)/tools/make-directive-index.py $(SOURCE
 EXTRA_DIST += \
 	man/systemd.index.xml \
 	man/index.html \
-	man/systemd.directives.xml
+	man/systemd.directives.xml \
+	man/glib-event-glue.c
 
 CLEANFILES += \
 	man/systemd.index.xml \
