From c71202228f31176eee2aa7c798c9f3ff681cf957 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Sat, 4 Oct 2014 13:11:40 +0200
Subject: [PATCH] systemd-bus-proxyd: distribute the .in file also for the user
 version

---
 Makefile.am | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index cca949feff..d2a1767265 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2477,13 +2477,16 @@ nodist_systemunit_DATA += \
 dist_systemunit_DATA += \
 	units/systemd-bus-proxyd.socket
 
-dist_userunit_DATA += \
-	units/user/systemd-bus-proxyd.socket \
+nodist_userunit_DATA += \
 	units/user/systemd-bus-proxyd@.service
+
+dist_userunit_DATA += \
+	units/user/systemd-bus-proxyd.socket
 endif
 
 EXTRA_DIST += \
-	units/systemd-bus-proxyd@.service.in
+	units/systemd-bus-proxyd@.service.in \
+	units/user/systemd-bus-proxyd@.service.in
 
 # ------------------------------------------------------------------------------
 systemd_tty_ask_password_agent_SOURCES = \
