From 402696d787d252d6317e36c5be340d47578bb27b Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 8 Oct 2014 21:32:30 +0200
Subject: [PATCH] Update TODO

---
 TODO | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/TODO b/TODO
index b437166273..7d205faa8d 100644
--- a/TODO
+++ b/TODO
@@ -34,6 +34,13 @@ External:
 
 Features:
 
+* bus-proxy: when passing messages from kdbus, make sure we properly
+  handle the case where a large number of fds is appended that we
+  cannot pass into sendmsg() of the AF_UNIX sokcet (which only accepts
+  253 messages)
+
+* check for validity of ucred fields in bus-proxy, like we do in journald
+
 * busctl: add a pcap writer, using LINKTYPE_DBUS/231
 
 * man: maybe use the word "inspect" rather than "introspect"?
