From 706d97503df83d141d241b645d2c920d691b3d62 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 9 Oct 2014 01:46:31 +0200
Subject: [PATCH] update TODO

---
 TODO | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/TODO b/TODO
index 7d205faa8d..45c08f0942 100644
--- a/TODO
+++ b/TODO
@@ -34,6 +34,10 @@ External:
 
 Features:
 
+* man: document that corrupted journal files is nothing to act on
+
+* systemctl isolate should use ".target" as completion suffix, not ".service"
+
 * bus-proxy: when passing messages from kdbus, make sure we properly
   handle the case where a large number of fds is appended that we
   cannot pass into sendmsg() of the AF_UNIX sokcet (which only accepts
