From 384a4be2b00cb95ce215dd343cc9aa77adc9b1ec Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 9 Oct 2014 16:51:31 +0200
Subject: [PATCH] update TODO

---
 TODO | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/TODO b/TODO
index 45c08f0942..dc20954582 100644
--- a/TODO
+++ b/TODO
@@ -34,6 +34,8 @@ External:
 
 Features:
 
+* maybe support a new very "soft" reboot mode, that simply kills all processes, disassembles everything, flushes /run and sysvipc, and then reexecs systemd again
+
 * man: document that corrupted journal files is nothing to act on
 
 * systemctl isolate should use ".target" as completion suffix, not ".service"
