Index: systemd-195/Makefile.in
===================================================================
--- systemd-195.orig/Makefile.in
+++ systemd-195/Makefile.in
@@ -83,7 +83,7 @@ noinst_PROGRAMS = test-engine$(EXEEXT) t
 	test-unit-file$(EXEEXT) test-date$(EXEEXT) \
 	test-libudev$(EXEEXT) test-udev$(EXEEXT) test-id128$(EXEEXT) \
 	test-journal$(EXEEXT) test-journal-send$(EXEEXT) \
-	test-journal-syslog$(EXEEXT) test-journal-match$(EXEEXT) \
+	test-journal-match$(EXEEXT) \
 	test-journal-stream$(EXEEXT) test-journal-verify$(EXEEXT) \
 	test-mmap-cache$(EXEEXT) $(am__EXEEXT_4)
 TESTS = test-job-type$(EXEEXT) test-env-replace$(EXEEXT) \
@@ -91,7 +91,7 @@ TESTS = test-job-type$(EXEEXT) test-env-
 	test-unit-file$(EXEEXT) test-date$(EXEEXT) test/udev-test.pl \
 	test/rules-test.sh $(am__append_40) test-id128$(EXEEXT) \
 	test-journal$(EXEEXT) test-journal-send$(EXEEXT) \
-	test-journal-syslog$(EXEEXT) test-journal-match$(EXEEXT) \
+	test-journal-match$(EXEEXT) \
 	test-journal-stream$(EXEEXT) test-journal-verify$(EXEEXT) \
 	test-mmap-cache$(EXEEXT)
 udevlibexec_PROGRAMS = ata_id$(EXEEXT) cdrom_id$(EXEEXT) \
