diff -up systemd-11/src/ask-password-agent.vala.notify systemd-11/src/ask-password-agent.vala
--- systemd-11/src/ask-password-agent.vala.notify	2010-11-12 20:27:39.535719001 -0500
+++ systemd-11/src/ask-password-agent.vala	2010-11-12 20:27:52.233719001 -0500
@@ -181,8 +181,7 @@ public class MyStatusIcon : StatusIcon {
 
                 set_visible(true);
 
-                Notification n = new Notification(title, message, icon, null);
-                n.attach_to_status_icon(this);
+                Notification n = new Notification(title, message, icon);
                 n.set_timeout(5000);
                 n.show();
 
