diff -up systemd-11/src/ask-password-agent.vala.vala-build systemd-11/src/ask-password-agent.vala
--- systemd-11/src/ask-password-agent.vala.vala-build	2010-11-12 20:51:53.579719000 -0500
+++ systemd-11/src/ask-password-agent.vala	2010-11-12 20:52:06.288719002 -0500
@@ -225,7 +225,7 @@ public class MyStatusIcon : StatusIcon {
 
                 OutputStream stream = new UnixOutputStream(to_process, true);
 
-                stream.write(password, password.length, null);
+                stream.write(password.data, null);
         }
 }
 
